demo/src/main/java/com/restaurant/demo/service/ProductService.java

```diff
@@ -12,14 +13,18 @@
 public class ProductService {
 	
 	@Autowired
-	ProductRepository productRepository;
+	private ProductRepository productRepository;
 	
 	public ArrayList<Product> listProducts(){
 		return (ArrayList<Product>) productRepository.findAll();
 	}
 	
-	public ArrayList<Product> listProductsByCategory(int id){
-		return (ArrayList<Product>) productRepository.listProductsByCategory(id);
+	public ArrayList<Product> listProductsByCategory(String desc){
```

Comment:
please return List instead refer to => https://stackoverflow.com/questions/11683044/return-arraylist-or-list#:~:text=Returning%20List%20will%20make%20it,them%20to%20use%20an%20ArrayList.

do the change in all the places you use ArrayList when returning a list.